Gems/ScriptCanvasTesting/Code/Tests/ScriptCanvas_Slots.cpp

```diff
@@ -286,6 +290,194 @@ TEST_F(ScriptCanvasTestFixture, SlotConnecting_ExecutionBasic)
     TestConnectionBetween(outputEndpoint, inputEndpoint, true);
 }
 
+// Test implicit connections against a simple node that has no slot execution map
+TEST_F(ScriptCanvasTestFixture, SlotConnecting_ImplicitConnections)
+{
+    using namespace ScriptCanvas;
+
+    auto editorGraph = CreateEditorGraph();
+
+    // Node before node that creates implicit connections
+    ConfigurableUnitTestNode* sourceNode = CreateConfigurableNode();
+    Endpoint execOutSlot = sourceNode->AddTestingSlot(CommonSlots::Execution("Out", ConnectionType::Output))->GetEndpoint();
+    Endpoint dataOutSlot1 = sourceNode->AddTestingSlot(CommonSlots::FloatData("out1", ConnectionType::Output))->GetEndpoint();
+    Endpoint dataOutSlot2 = sourceNode->AddTestingSlot(CommonSlots::FloatData("out2", ConnectionType::Output))->GetEndpoint();
+
+    // Node that creates implicit connections
+    ConfigurableUnitTestNode* targetNode = CreateConfigurableNode();
+    Endpoint implicitSlot = targetNode->AddTestingSlot(CommonSlots::Execution("In", ConnectionType::Input, false, true))->GetEndpoint();
+    Endpoint dataInSlot1 = targetNode->AddTestingSlot(CommonSlots::FloatData("in1", ConnectionType::Input))->GetEndpoint();
+    Endpoint dataInSlot2 = targetNode->AddTestingSlot(CommonSlots::FloatData("in2", ConnectionType::Input))->GetEndpoint();
+
+    // Test the implicit connections between the two nodes
+    TestAllImplicitConnections(editorGraph, { dataOutSlot1, dataOutSlot2 }, { dataInSlot1, dataInSlot2 }, execOutSlot, implicitSlot, { execOutSlot });
+}
+
+// Test implicit connections against a complex node that has a slot execution map
+TEST_F(ScriptCanvasTestFixture, SlotConnecting_ImplicitConnectionsSlotExecutionMap)
+{
+    using namespace ScriptCanvas;
+
+    auto editorGraph = CreateEditorGraph();
+
+    // These vectors store each "set" of data output endpoints that correspond with one execution out
+    AZStd::vector<Endpoint> dataOutEndpointSet1;
+    AZStd::vector<Endpoint> dataOutEndpointSet2;
+    AZStd::vector<Endpoint> dataOutEndpointSet3;
+    AZStd::vector<Endpoint> dataOutEndpointSet4;
+    AZStd::vector<Endpoint> dataOutEndpointSet5;
```

Comment:
I see the comments below explaining the scenario these sets are used for testing. Might be worth adding a little blurb here for them as well so folks can see right away what these are used for, since set1/set2/etc... isn't prescriptive on it's own.